crates/tests/e2e-tests/main.rs

```diff
@@ -342,10 +345,20 @@ async fn test_eth_block_rpc(#[case] test_name: &str) {
                 panic!("Block receipts should be an array");
             }
 
+            let block = operations::eth_get_block_by_number_or_hash(
+                &client,
+                operations::BlockId::Number(target_block_number),
+                false,
+            )
+            .await
+            .expect("Failed to get block from client");
+            let block_hash =
+                block["hash"].as_str().expect("Block hash should not be empty").to_string();
+
             // Test getting block receipts by block hash
             let receipts_by_hash = operations::eth_get_block_receipts(
                 &client,
-                operations::BlockId::Hash(target_block_hash.clone()),
+                operations::BlockId::Hash(block_hash.clone()),
```

Comment:
I think you do not need to clone. I believe you can do `block_hash.into()` instead.
```suggestion
            let block_hash =
                block["hash"].as_str().expect("Block hash should not be empty");

// Test getting block receipts by block hash
let receipts_by_hash = operations::eth_get_block_receipts(
&client,
                operations::BlockId::Hash(block_hash.into()),
```

Reply:
Done.